### PR TITLE
Added remaining processing models

### PR DESCRIPTION
**Pull request by pmeenan**
Added the steps for HTML integration with the remaining tags, based on the steps used for refererpolicy (where appripriate).

This included the base integration with the HTML "Fetching Resources" section.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -216,16 +239,52 @@ urlPrefix: https://fetch.spec.whatwg.org; spec: FETCH;
...
+        </pre>
+      <li><p>We add to the <{img}> element content attribute descriptions:</p>
+        <p>The <dfn element-attr for="img"><code data-x="attr-img-importance">importance</code>
+          </dfn> attribute is an [=importance attribute=]. Its purpose is to set the [=importance=]
+          used when [=fetching=] the image.</p>
+        <p>The <dfn attribute for="HTMLImageElement"><code
+          data-x="dom-img-importance">importance</code></dfn> IDL attribute must
+          <span>reflect</span> the <code data-x="attr-img-importance">importance</code>
+          content attribute, <span>limited to only known values</span>.</p>
+      <li><p>We add to [=Reacting to DOM mutations=]:</p>
```
Comment (yoavweiss) [inline index.bs]: I don't think `importance` changes should be considered a relevant mutation, as it's not something that should e.g. trigger an image request if the image is already loaded.

File: index.bs [hunk 2]
```diff
@@ -237,41 +296,102 @@ urlPrefix: https://fetch.spec.whatwg.org; spec: FETCH;
...
             [CEReactions] attribute DOMString importance;
         };
         </pre>
-      <li>TODO: Complete documenting the various points in the link processing model to pass the
-        importance through to the fetch for the referenced link resource.
+      <li><p>We extend [=create a link element request=] of [=Fetching and processing a resource
+        from a link element=] by adding a step before returning the request:</p>
+        <p>Set <var>request</var>'s {{Request/importance}} to the current state of <var ignore=''>
+        el</var>'s <code data-x="attr-link-importance">importance</code> content attribute.</p>
+      <li><p>We extend the [=fetch and process the linked resource=] algorithm of [=modulepreload=]
```
Comment (yoavweiss) [inline index.bs:324]: Tangent: Weird that `modulepreload`'s is defined apart from other links...

File: index.bs [hunk 3]
```diff
@@ -237,41 +296,102 @@ urlPrefix: https://fetch.spec.whatwg.org; spec: FETCH;
...
+      <li>We extend the <{script}> element DOM interface as follows:
         <pre class="idl">
         partial interface HTMLScriptElement {
-            [CEReactions] attribute DOMString importance;
+            [CEReactions] attribute DOMString <span data-x="dom-script-importance">importance</script>;
         };
         </pre>
-      <li>TODO: Complete documenting the various points in the script processing model to pass the
-        importance through to the fetch for the referenced script resource.
+      <li><p>We add to the <{img}> element content attribute descriptions:</p>
```
Comment (yoavweiss) [inline index.bs]: s/img/script/ ?

File: index.bs [hunk 4]
```diff
@@ -237,41 +296,102 @@ urlPrefix: https://fetch.spec.whatwg.org; spec: FETCH;
...
+        <ol>
+          <li>The <dfn element-attr for="script"><code data-x="attr-script-importance">importance
+            </code></dfn> attribute is an [=importance attribute=]. Its purpose is to set the
+            [=importance=] used when [=fetching=] the script.
+          <li>The <dfn attribute for="HTMLScriptElement"><code
+            data-x="dom-script-importance">importance</code></dfn> IDL attribute must
+            <span>reflect</span> the <code data-x="attr-script-importance">importance</code>
+            content attribute, <span>limited to only known values</span>.
+          <li>We add <code data-x="attr-script-importance">importance</code> to the list of
+            attributes that have no direct effect when changed dynamically.
```
Comment (yoavweiss) [inline index.bs]: Can you clarify what you mean "list of attributes that have no direct effect.."? Can you link to that?

File: index.bs [hunk 5]
```diff
@@ -237,41 +296,102 @@ urlPrefix: https://fetch.spec.whatwg.org; spec: FETCH;
...
+        </ol>
+      <li>We modify the list of attributes not to be specified when used in [=data blocks=] to
+        include the <code data-x="attr-script-importance">importance</code> attribute.
```
Comment (yoavweiss) [inline index.bs:358]: Can you link to this list of attributes?

File: index.bs [hunk 6]
```diff
@@ -237,41 +296,102 @@ urlPrefix: https://fetch.spec.whatwg.org; spec: FETCH;
...
+      <li><p>We extend the [=Script fetch options=] struct:<p>
+        <p><span data-x="concept-script-fetch-options-importance">importance</span> - The
+          {{Request/importance}} for the initial [=fetch=].</p>
+      <li>We extend [=default classic script fetch options=] to include "<code>auto</code>"
+        as the default value for <code data-x="concept-script-fetch-options-importance">
+        importance</code>.
+      <li>We extend [=set up the classic script request=] to set <var>request</var>'s
+        {{Request/importance}} to <var>option</var>'s
+        <code data-x="concept-script-fetch-options-importance">importance</code>
+      <li>We extend [=set up the module script request=] to set <var>request</var>'s
```
Comment (yoavweiss) [inline index.bs:374]: I think you're missing setting the importance on the options in step 24 of https://html.spec.whatwg.org/multipage/scripting.html#prepare-a-script

File: index.bs [hunk 7]
```diff
@@ -237,41 +296,102 @@ urlPrefix: https://fetch.spec.whatwg.org; spec: FETCH;
...
-        importance through to the fetch for the referenced iframe resource.
+      <li><p>We add to the <{iframe}> content attribute descriptions:</p>
+        <p>The <dfn element-attr for="iframe"><code data-x="attr-iframe-importance">importance
+          </code></dfn> attribute is an [=importance attribute=]. Its purpose is to set the
+          [=importance=] used when [=processing the iframe attributes=].</p>
+        <p>The <dfn attribute for="HTMLIFrameElement"><code
+          data-x="dom-iframe-importance">importance</code></dfn> IDL attribute must
+          <span>reflect</span> the <code data-x="attr-iframe-importance">importance</code>
+          content attribute, <span>limited to only known values</span>.</p>
+      <li><p>We modify the resource creation step of
```
Comment (yoavweiss) [inline index.bs]: Worth clarifying what you mean by "reasource creation step" (which I think is step 5 atm)

**Review by yoavweiss — COMMENTED** [review]
Looks great!! A few comments.

File: index.bs (hunk 1 above)
Comment (pmeenan) [reply index.bs]: removed, thx.

File: index.bs (hunk 3 above)
Comment (pmeenan) [reply index.bs]: Whoops, thanks. Copy/paste caught up with me.

File: index.bs (hunk 5 above)
Comment (pmeenan) [reply index.bs:358]: the #data-block link was as close as I could get without using a text-based link. It takes you to the section that includes the list of attributes. It's not linkable directly and doesn't have a number:

Screen Shot 2022-02-23 at 4 09 45 PM
.

File: index.bs (hunk 4 above)
Comment (pmeenan) [reply index.bs]: Sorry, there's this paragraph as part of the description of the script element but it's not in a section that is directly linkable.  Is there a good way to provide better context?
Screen Shot 2022-02-23 at 4 13 16 PM

File: index.bs (hunk 6 above)
Comment (pmeenan) [reply index.bs:374]: Thanks, done.

File: index.bs (hunk 7 above)
Comment (pmeenan) [reply index.bs]: Done

**Comment by pmeenan** [comment]
@yoavweiss I think most of the issues are addressed now (thanks). The 2 I'm having issues with are trying to deep-link into the 2 paragraphs in the script element description that list attributes (dynamic and data block) that aren't numbered steps or directly linkable.

File: index.bs (hunk 4 above)
Comment (yoavweiss) [reply index.bs]: The downsides of monkeypatching... Maybe a link-to-text, which falls back on the section, e.g. https://html.spec.whatwg.org/multipage/scripting.html#:~:text=has%20no%20direct%20effect (with a note telling folks with non-supporting browsers to search for "has no effect")

File: index.bs (hunk 5 above)
Comment (yoavweiss) [reply index.bs:358]: OK, it might be better to directly link to https://html.spec.whatwg.org/multipage/scripting.html#the-script-element:data-block-2

**Review by yoavweiss — APPROVED** [review]
LGTM % linking nits

Thanks!! :)

**Pull request merged**